### PR TITLE
rekor-cli, rekor-server: 0.1.1 -> 0.2.0

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Motivation for this change

https://github.com/sigstore/rekor/releases/tag/v0.2.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 127191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>rekor-cli</li>
    <li>rekor-server</li>
  </ul>
</details>

**Review by 06kellyjac — APPROVED** [review]
LGTM

File: pkgs/tools/security/rekor/default.nix [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
 
-      vendorSha256 = "0vdir9ia3hv27rkm6jnvhsfc3mxw36xfvwqnfd34rgzmzcfxlrbv";
+      vendorSha256 = "1wlh505ypwyr91wi80fpbap3far3fljwjd4mql2qcqgg0b1yay9s";
 
-      inherit subPackages postInstall;
+      subPackages = [ packageToBuild ];
+
+      preBuild = ''
+        buildFlagsArray+=("-ldflags" "-s -w -X github.com/sigstore/rekor/${packageToBuild}/app.gitVersion=v${version}")
+      '';
```
Comment (SuperSandro2000) [inline pkgs/tools/security/rekor/default.nix:22]: Please use ldflags array.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/rekor/default.nix (hunk 1 above)
Comment (06kellyjac) [reply pkgs/tools/security/rekor/default.nix:22]: you want it moved back to this format??

```
buildFlagsArray = [
  "-ldflags="
  "-s"
  "-w"
  "-X github.com/sigstore/rekor/${packageToBuild}/app.gitVersion=v${version}"
];
```

Comment (SuperSandro2000) [reply pkgs/tools/security/rekor/default.nix:22]: We have something new and fancy https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#ldflags-var-go-ldflags

Comment (06kellyjac) [reply pkgs/tools/security/rekor/default.nix:22]: nice, I'll make sure to change my packages next time I update them :+1: 

Is there a way to find out about these kinds of changes other than word-of-mouth?

Comment (SuperSandro2000) [reply pkgs/tools/security/rekor/default.nix:22]: > Is there a way to find out about these kinds of changes other than word-of-mouth?

I don't think so.

Comment (LeSuisse) [reply pkgs/tools/security/rekor/default.nix:22]: Thanks for the info :+1: . I will take care of it next time.

**Pull request merged**